codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/protocol/rpcv2cbor/generators/ErrorsGenerator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.ruby.codegen.protocol.rpcv2cbor.generators;
+
+
+import software.amazon.smithy.ruby.codegen.GenerationContext;
+import software.amazon.smithy.ruby.codegen.Hearth;
+import software.amazon.smithy.ruby.codegen.generators.ErrorsGeneratorBase;
+
+public class ErrorsGenerator extends ErrorsGeneratorBase {
+
+    public ErrorsGenerator(GenerationContext context) {
+        super(context);
+    }
+
+    @Override
+    public void renderErrorCodeBody() {
+        writer
+                .write("")
+                .openBlock("if !(200..299).cover?(resp.status)")
+                .write("data = $T.decode(resp.body.read.force_encoding(Encoding::BINARY))", Hearth.CBOR)
```

Comment:
I think all of the force_encoding(encoding::binary) should be moved to the decode method in hearth CBOR. That way it's not in all of the codegen parsers. I think it's reasonable for the decoder to try and coerce it into what it expects to decode if it hasn't been already.

Reply:
Yeah - I think that actually makes sense.  I was initially trying to avoid baking the cost of this encoding into the decode method but looking at the actual C implementation, if the encoding already matches, it exits immediately and doesn't copy memory - so I think its reasonable to push down to the decoder.